zipline/finance/performance/position_tracker.py

```diff
@@ -64,6 +71,72 @@ def _update_multipliers(self, sid):
                     asset.contract_multiplier
                 self._position_payout_multipliers[sid] = \
                     asset.contract_multiplier
+                # Futures are closed on their notice_date
+                if asset.notice_date:
+                    self._insert_auto_close_position_date(
+                        dt=asset.notice_date,
+                        sid=sid
+                    )
+                # If the Future does not have a notice_date, it will be closed
+                # on its expiration_date
+                elif asset.expiration_date:
+                    self._insert_auto_close_position_date(
+                        dt=asset.expiration_date,
+                        sid=sid
+                    )
+
+    def _insert_auto_close_position_date(self, dt, sid):
+        """
+        Inserts the given SID in to the list of positions to be auto-closed by
+        the given dt.
+
+        Parameters
+        ----------
+        dt : pandas.Timestamp
+            The date before-which the given SID will be auto-closed
+        sid : int
+            The SID of the Asset to be auto-closed
+        """
+        if dt not in self._auto_close_position_sids:
+            bisect.insort(self._auto_close_position_dates, dt)
+        self._auto_close_position_sids.setdefault(dt, set()).add(sid)
```

Comment:
The catch with an OrderedDict is that it maintains order of insertion, and the only way to re-order is to re-build the dict entirely. I figured that was too expensive for our purposes, so keeping a separate, ordered, insertable list seemed the most elegant way to keep the iteration ordered.

Is there a more elegant way to re-tool this, or do you think this one is ready to go?